### PR TITLE
#101: Duplicate keys in bc file incorrectly leads to semicolon separated keys

### PR DESCRIPTION
**Pull request by priscavdsluis**

### CODE REVIEW

**Comment by evetion** [comment]
Interesting, there seem to be unicode issue on Windows

File: hydrolib/core/io/bc/models.py [hunk 1]
```diff
@@ -37,15 +33,24 @@ class TimeInterpolation(str, Enum):
     block_to = "blockTo"
 
 
+class QuantityUnitPair(NamedTuple):
+    quantity: str
+    unit: str
+
+    def to_properties(self):
```
Comment (evetion) [inline hydrolib/core/io/bc/models.py:40]: Could be `_to_properties`?

**Review by evetion — APPROVED** [review]
Nice code. Could you provide a test example, or documentation example where you intialize the quantities with keys? Now it could be unclear, I rather have:

```python
       quantities=[
            (quantity="qhbnd discharge", unit="m3/s"),
            (quantity="qhbnd waterlevel", unit="m"),
        ],
```

Than the unnamed, current, solution:
```python
       quantities=[
            ("qhbnd discharge", "m3/s"),
            ("qhbnd waterlevel", "m"),
        ],
```

File: tests/io/test_bc.py [hunk 2]
```diff
@@ -98,7 +106,7 @@ class TestForcingBase:
         ],
     )
     def test_parses_function_case_insensitive(self, input, expected):
-        forcing = ForcingBase(function=input, name="somename", quantity=[], unit=[])
+        forcing = ForcingBase(function=input, name="somename", quantities=[])
```
Comment (arthurvd) [inline tests/io/test_bc.py:109]: In some earlier commit, apparently ```quantity``` was renamed to ```quantities```.
This intuitive, but it is inconsistent with class members in many other classes, for example: ```StructureModel.structure```, ```CrossDefModel.definition```, etc., which are all also lists.
Not sure what is best to change or keep here...

**Review by arthurvd — APPROVED** [review]

File: tests/io/test_bc.py (hunk 2 above)
Comment (priscavdsluis) [reply tests/io/test_bc.py]: Changed the name as agreed to `quantityunitpair`

**Review by arthurvd — APPROVED** [review]

**Pull request merged**